liquidity/liquidity.go

```diff
@@ -810,7 +824,18 @@ func (m *Manager) suggestSwap(ctx context.Context, traffic *swapTraffic,
 		return nil, ReasonLiquidityOk, nil
 	}
 
-	return m.loopOutSwap(ctx, amount, balance, autoloop)
+	swap, reason, err := m.loopOutSwap(ctx, amount, balance, autoloop)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	if reason != ReasonNone {
```

Comment:
+1

Reply:
My reluctance here is that I'd need to map errors to reasons 1:1 to do so, which seems like an unnecessary layer, but will give it a try and see what it looks like.

Reply:
Can't you just use an error type that has the reason embedded and then use `errors.As()` to get the reason out?

Reply:
Done, awesome suggestion 🎉 Haven't used `errors.As` before, very cool!

Diff since last review is [here](https://github.com/lightninglabs/loop/compare/f0a3b9375e9273a87a77eb907ba15964a507c011..62e8063300292074aa2161cc4e2e892de6ecd1f0).